Assets/Scripts/NPC/Waypoint.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Waypoint : MonoBehaviour
+{
+    [SerializeField] private Grid parentGrid;
+    [SerializeField] private bool isOccupied;
+    [SerializeField] private float connectedWaypointRadius;
+    [SerializeField] private Collider2D[] hits = new Collider2D[7];
+    [SerializeField] private List<Waypoint> connectedWaypoints = new List<Waypoint>();
+
+    [SerializeField] private bool drawGizmos;
+
+    private Waypoint connectedWaypoint;
+    public bool IsOccupied => isOccupied;
```

Comment:
```suggestion
    [field: SerializeField] private bool IsOccupied { get; private set; }
    [SerializeField] private float connectedWaypointRadius;
    [SerializeField] private Collider2D[] hits = new Collider2D[7];
    [SerializeField] private List<Waypoint> connectedWaypoints = new List<Waypoint>();

    [SerializeField] private bool drawGizmos;

    private Waypoint connectedWaypoint;
```
Je kan de IsOccupied bool veranderen naar een get en private set met field er bij om het nog steeds in de inspector te zien.